In [1]:
import pandas as pd
import numpy as np
import ast


In [2]:
annotated_comment_level = pd.read_csv('annotated_comment_level.csv')
annotated_issue_level= pd.read_csv('annotated_issue_level.csv')
comments = pd.read_csv('comments.csv')
issue_threads = pd.read_csv('issue_threads.csv')

In [3]:
annotated_comment_level

id    issue_id  comment_id                           tbdf  \
0        1     6209234     6209234                           None   
1        2     6209234     7718792                    Entitlement   
2        3     6209234     7722530                           None   
3        4     6209234     7722736                           None   
4        5     6209234     7744377                     Impatience   
...    ...         ...         ...                            ...   
5964  5965  1958701035  1780979523  Identity attacks/Name-Calling   
5965  5966  1958701035  1958701035                           None   
5966  5967  1959812102  1777791158                           None   
5967  5968  1959812102  1777796247                      Insulting   
5968  5969  1959812102  1959812102                           None   

                                           comment_body  
0     I noticed that Prepared Statements seem to be ...  
1     Yes, prepared statements are on my todo list. ...  
2     The only downside with the SQL-based approach ...  
3     Prepared statements use a range of additional ...  
4     How does https://github.com/sidorares/nodejs-m...  
...                                                 ...  
5964  Hey dude, when Palestinians go to Israel and b...  
5965                                                NaN  
5966               No, this is grammatically incorrect.  
5967  What the heck is going on here with all these ...  
5968                                                NaN  

[5969 rows x 5 columns]

In [4]:
annotated_issue_level

id    issue_id                                    trigger      target  \
0      1   774727384                                       None      People   
1      2   654775761                                       None      People   
2      3  1418577350                                       None   Code/tool   
3      4  1345626221                    Communication breakdown      People   
4      5   723058396                    Communication breakdown      People   
..   ...         ...                                        ...         ...   
399  401  2044991706  Failed use of tool/code or error messages      People   
400  402  1444401730  Failed use of tool/code or error messages      People   
401  403  1458781974                    Communication breakdown  Undirected   
402  404  1473837925                                       None        None   
403  405  1498285225                                       None        None   

                                          consequences  
0    ['Provided technical explanation', 'Discontinu...  
1                  ['Discontinued further discussion']  
2                               ['Escalating further']  
3                  ['Discontinued further discussion']  
4    ['Escalating further', 'Trying to stop the inc...  
..                                                 ...  
399                                           ['None']  
400                 ['Provided technical explanation']  
401                                           ['None']  
402                                           ['None']  
403                                           ['None']  

[404 rows x 5 columns]

In [5]:
comments

Unnamed: 0              project_name   issue_id  comment_id  \
0              0               rails/rails  352750519   352750519   
1              1               rails/rails  352750519   414870475   
2              2               rails/rails  352750519   414873068   
3              3               rails/rails  352750519   414873738   
4              4               rails/rails  352750519   414875618   
...          ...                       ...        ...         ...   
7171        7171  Homebrew/homebrew-bundle  276835188   362543755   
7172        7172  Homebrew/homebrew-bundle  276835188   362582972   
7173        7173  Homebrew/homebrew-bundle  276835188   362606705   
7174        7174  Homebrew/homebrew-bundle  276835188   363021802   
7175        7175  Homebrew/homebrew-bundle  276835188   363022023   

                                           comment_body            created_at  \
0     Per https://twitter.com/dhh/status/10320503255...  2018-08-21T23:48:12Z   
1     Good intentions, but I doubt there's any relat...  2018-08-22T00:57:07Z   
2     At least [one source from a search](https://ww...  2018-08-22T01:11:49Z   
3     Regardless of origin, allow/deny are simply cl...  2018-08-22T01:15:48Z   
4     1. [etymology is quite important](\r\nhttps://...  2018-08-22T01:26:22Z   
...                                                 ...                   ...   
7171  @paulp I cannot reproduce this locally with th...  2018-02-02T10:10:03Z   
7172  Re #321, it's a different bug - as I said, "th...  2018-02-02T13:14:34Z   
7173  @paulp Calling someone who maintains software ...  2018-02-02T14:50:05Z   
7174               @paulp I'll take that as a no, then.  2018-02-05T09:05:33Z   
7175  Anyone else who hits this: we'll consider acce...  2018-02-05T09:06:22Z   

       user_id  
0         2741  
1     14361073  
2        10137  
3         2741  
4       201135  
...        ...  
7171    125011  
7172      9135  
7173    125011  
7174    125011  
7175    125011  

[7176 rows x 7 columns]

In [6]:
issue_threads

issue_id  total_comments  \
0      12894489               6   
1      13258430              27   
2      22814728               4   
3      24996968              13   
4      27120442              16   
..          ...             ...   
399  1897778633               2   
400  1906791437               2   
401  1943190932               7   
402  1958701035               5   
403  1959812102               3   

                                                   url  \
0    https://api.github.com/repos/doctrine/mongodb-...   
1    https://api.github.com/repos/capistrano/capist...   
2    https://api.github.com/repos/Leaflet/Leaflet/i...   
3    https://api.github.com/repos/composer/composer...   
4    https://api.github.com/repos/mysqljs/mysql/iss...   
..                                                 ...   
399  https://api.github.com/repos/FreeRADIUS/freera...   
400  https://api.github.com/repos/mozilla/pdf.js/is...   
401  https://api.github.com/repos/php/php-src/issue...   
402  https://api.github.com/repos/composer/packagis...   
403  https://api.github.com/repos/factor/factor/iss...   

                                           issue_title  
0          References to embedded document collections  
1    reading the variable "latest_release" before "...  
2                      Leaflet JS - Country Boundaries  
3                 Support for users without SSH access  
4                  Security issue with param escaping?  
..                                                 ...  
399                  Question about fixing memory leak  
400  Could you please explain why the default full ...  
401     Better name for method Randomizer::nextFloat()  
402     Add #StandWithPalestine banner to updates ____  
403                                   Update README.md  

[404 rows x 4 columns]

In annotated_issue_level, we add a new column, consequence_category, showing the predominant consequence (positive, negative, neutral). We then make a new dataframe, basically adding to each row from annotated_comment_level, the issue trigger, target and consequence_category. This will be done merging on issue_id. This format will then enable nesting at issue_id level in STATA.

In [7]:
annotated_issue_level.rename(columns={' consequences': 'consequences'}, inplace=True)
annotated_issue_level['consequences'] = annotated_issue_level['consequences'].apply(ast.literal_eval)

In [8]:
sentiment_map = {
    'Provided technical explanation': 'positive',
    'Discontinued further discussion': 'neutral',
    'Accepting criticism': 'positive',
    'Escalating further': 'negative',
    'Trying to stop the incivility': 'neutral',
    'None': 'neutral',
    'Invoke Code of Conduct': 'negative'
}

In [9]:
annotated_issue_level['positive'] = annotated_issue_level['consequences'].apply(
    lambda row: sum(sentiment_map[item] == 'positive' for item in row)
)
annotated_issue_level['negative'] = annotated_issue_level['consequences'].apply(
    lambda row: sum(sentiment_map[item] == 'negative' for item in row)
)
annotated_issue_level['neutral'] = annotated_issue_level['consequences'].apply(
    lambda row: sum(sentiment_map[item] == 'neutral' for item in row)
)

In [10]:
annotated_issue_level

id    issue_id                                    trigger      target  \
0      1   774727384                                       None      People   
1      2   654775761                                       None      People   
2      3  1418577350                                       None   Code/tool   
3      4  1345626221                    Communication breakdown      People   
4      5   723058396                    Communication breakdown      People   
..   ...         ...                                        ...         ...   
399  401  2044991706  Failed use of tool/code or error messages      People   
400  402  1444401730  Failed use of tool/code or error messages      People   
401  403  1458781974                    Communication breakdown  Undirected   
402  404  1473837925                                       None        None   
403  405  1498285225                                       None        None   

                                          consequences  positive  negative  \
0    [Provided technical explanation, Discontinued ...         2         0   
1                    [Discontinued further discussion]         0         0   
2                                 [Escalating further]         0         1   
3                    [Discontinued further discussion]         0         0   
4    [Escalating further, Trying to stop the incivi...         0         1   
..                                                 ...       ...       ...   
399                                             [None]         0         0   
400                   [Provided technical explanation]         1         0   
401                                             [None]         0         0   
402                                             [None]         0         0   
403                                             [None]         0         0   

     neutral  
0          1  
1          1  
2          0  
3          1  
4          1  
..       ...  
399        1  
400        0  
401        1  
402        1  
403        1  

[404 rows x 8 columns]

In [11]:
# Function to determine the category
def get_consequence_category(row):
    if row['positive'] > row['negative']:
        return 'positive'
    elif row['negative'] > row['positive']:
        return 'negative'
    else:
        # Check if both are zero
        if row['positive'] == 0 and row['negative'] == 0:
            return 'neutral'
        else:
            return 'balanced'

In [12]:
annotated_issue_level['consequence_category'] = annotated_issue_level.apply(get_consequence_category, axis=1)

In [13]:
annotated_issue_level['consequence_category'].value_counts()

neutral     245
negative     88
positive     55
balanced     16
Name: consequence_category, dtype: int64

In [14]:
annotated_comment_level['tbdf'] = annotated_comment_level['tbdf'].replace(
    'Identity Attack/Name-Calling', 'Identity attacks/Name-Calling'
)

In [15]:
annotated_comment_level['tbdf'] = annotated_comment_level['tbdf'].replace(
    'Identity attack/Name-Calling', 'Identity attacks/Name-Calling'
)

In [16]:
annotated_comment_level['tbdf'].value_counts()

None                             4599
Bitter frustration                493
Impatience                        266
Mocking                           181
Insulting                         174
Vulgarity                          71
Entitlement                        70
Irony                              64
Identity attacks/Name-Calling      28
Threat                             23
Name: tbdf, dtype: int64

In [17]:
comment_issue_merge = pd.merge(annotated_comment_level, annotated_issue_level, on='issue_id', how='left')

In [18]:
comment_issue_merge

id_x    issue_id  comment_id                           tbdf  \
0        1     6209234     6209234                           None   
1        2     6209234     7718792                    Entitlement   
2        3     6209234     7722530                           None   
3        4     6209234     7722736                           None   
4        5     6209234     7744377                     Impatience   
...    ...         ...         ...                            ...   
5964  5965  1958701035  1780979523  Identity attacks/Name-Calling   
5965  5966  1958701035  1958701035                           None   
5966  5967  1959812102  1777791158                           None   
5967  5968  1959812102  1777796247                      Insulting   
5968  5969  1959812102  1959812102                           None   

                                           comment_body  id_y    trigger  \
0     I noticed that Prepared Statements seem to be ...   NaN        NaN   
1     Yes, prepared statements are on my todo list. ...   NaN        NaN   
2     The only downside with the SQL-based approach ...   NaN        NaN   
3     Prepared statements use a range of additional ...   NaN        NaN   
4     How does https://github.com/sidorares/nodejs-m...   NaN        NaN   
...                                                 ...   ...        ...   
5964  Hey dude, when Palestinians go to Israel and b...  97.0  Rejection   
5965                                                NaN  97.0  Rejection   
5966               No, this is grammatically incorrect.  12.0       None   
5967  What the heck is going on here with all these ...  12.0       None   
5968                                                NaN  12.0       None   

                    target          consequences  positive  negative  neutral  \
0                      NaN                   NaN       NaN       NaN      NaN   
1                      NaN                   NaN       NaN       NaN      NaN   
2                      NaN                   NaN       NaN       NaN      NaN   
3                      NaN                   NaN       NaN       NaN      NaN   
4                      NaN                   NaN       NaN       NaN      NaN   
...                    ...                   ...       ...       ...      ...   
5964  Company/organization  [Escalating further]       0.0       1.0      0.0   
5965  Company/organization  [Escalating further]       0.0       1.0      0.0   
5966                People                [None]       0.0       0.0      1.0   
5967                People                [None]       0.0       0.0      1.0   
5968                People                [None]       0.0       0.0      1.0   

     consequence_category  
0                     NaN  
1                     NaN  
2                     NaN  
3                     NaN  
4                     NaN  
...                   ...  
5964             negative  
5965             negative  
5966              neutral  
5967              neutral  
5968              neutral  

[5969 rows x 13 columns]

In [19]:
comment_issue_merge['consequence_category'].isna().sum()

14

We drop the 14 comments that do not belong to any of the annotated issues. 

In [20]:
comment_issue_merge = comment_issue_merge.dropna(subset=["consequence_category"])

In [21]:
comment_issue_merge

id_x    issue_id  comment_id                           tbdf  \
8        9    12894489    12894489                           None   
9       10    12894489    19156555                           None   
10      11    12894489   913841398                           None   
11      12    12894489   913845504                          Irony   
12      13    12894489   914030250                          Irony   
...    ...         ...         ...                            ...   
5964  5965  1958701035  1780979523  Identity attacks/Name-Calling   
5965  5966  1958701035  1958701035                           None   
5966  5967  1959812102  1777791158                           None   
5967  5968  1959812102  1777796247                      Insulting   
5968  5969  1959812102  1959812102                           None   

                                           comment_body   id_y    trigger  \
8     I'm not sure if this is officially supported b...  231.0  Rejection   
9     This doesn't work if class `B` has a binary ID...  231.0  Rejection   
10    This issue has been automatically marked as st...  231.0  Rejection   
11    Imagine thinking anyone cares about design flaws.  231.0  Rejection   
12    Imagine thinking anyone cares about Open Sourc...  231.0  Rejection   
...                                                 ...    ...        ...   
5964  Hey dude, when Palestinians go to Israel and b...   97.0  Rejection   
5965                                                NaN   97.0  Rejection   
5966               No, this is grammatically incorrect.   12.0       None   
5967  What the heck is going on here with all these ...   12.0       None   
5968                                                NaN   12.0       None   

                    target          consequences  positive  negative  neutral  \
8     Company/organization  [Escalating further]       0.0       1.0      0.0   
9     Company/organization  [Escalating further]       0.0       1.0      0.0   
10    Company/organization  [Escalating further]       0.0       1.0      0.0   
11    Company/organization  [Escalating further]       0.0       1.0      0.0   
12    Company/organization  [Escalating further]       0.0       1.0      0.0   
...                    ...                   ...       ...       ...      ...   
5964  Company/organization  [Escalating further]       0.0       1.0      0.0   
5965  Company/organization  [Escalating further]       0.0       1.0      0.0   
5966                People                [None]       0.0       0.0      1.0   
5967                People                [None]       0.0       0.0      1.0   
5968                People                [None]       0.0       0.0      1.0   

     consequence_category  
8                negative  
9                negative  
10               negative  
11               negative  
12               negative  
...                   ...  
5964             negative  
5965             negative  
5966              neutral  
5967              neutral  
5968              neutral  

[5955 rows x 13 columns]

In [22]:
#comment_issue_merge.to_csv('comment_issue_merge.csv', index=False)

In [23]:
comment_issue_merge['issue_id'].value_counts().mean()

14.77667493796526

5955 comments, from 403 issue threads. Issue threads have from 1 to 183 comments, with a mean value of 14.77 comments/thread.

In [24]:
comment_issue_merge.rename(columns={'tbdf': 'comment_label'}, inplace=True)
comment_issue_merge['entitlement']=0
comment_issue_merge['identity'] = 0
comment_issue_merge['impatience'] = 0
comment_issue_merge['insulting'] = 0
comment_issue_merge['none'] = 0
comment_issue_merge['irony'] = 0
comment_issue_merge['mocking'] = 0
comment_issue_merge['threat'] = 0
comment_issue_merge['vulgarity'] = 0
comment_issue_merge.loc[comment_issue_merge['comment_label'] == 'Entitlement', 'entitlement'] = 1
comment_issue_merge.loc[comment_issue_merge['comment_label'] == 'Identity attacks/Name-Calling', 'identity'] = 1
comment_issue_merge.loc[comment_issue_merge['comment_label'] == 'Impatience', 'impatience'] = 1
comment_issue_merge.loc[comment_issue_merge['comment_label'] == 'Insulting', 'insulting'] = 1
comment_issue_merge.loc[comment_issue_merge['comment_label'] == 'None', 'none'] = 1
comment_issue_merge.loc[comment_issue_merge['comment_label'] == 'Irony', 'irony'] = 1
comment_issue_merge.loc[comment_issue_merge['comment_label'] == 'Mocking', 'mocking'] = 1
comment_issue_merge.loc[comment_issue_merge['comment_label'] == 'Threat', 'threat'] = 1
comment_issue_merge.loc[comment_issue_merge['comment_label'] == 'Vulgarity', 'vulgarity'] = 1

C:\Users\20200604\AppData\Local\Temp\ipykernel_18036\4049301760.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  comment_issue_merge.rename(columns={'tbdf': 'comment_label'}, inplace=True)
C:\Users\20200604\AppData\Local\Temp\ipykernel_18036\4049301760.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  comment_issue_merge['entitlement']=0
C:\Users\20200604\AppData\Local\Temp\ipykernel_18036\4049301760.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See t

In [25]:
comment_issue_merge['entitlement'].value_counts()

0    5887
1      68
Name: entitlement, dtype: int64

In [26]:
comment_issue_merge['identity'].value_counts()

0    5927
1      28
Name: identity, dtype: int64

In [27]:
comment_issue_merge['impatience'].value_counts()

0    5692
1     263
Name: impatience, dtype: int64

In [28]:
comment_issue_merge['insulting'].value_counts()

0    5781
1     174
Name: insulting, dtype: int64

In [29]:
comment_issue_merge['none'].value_counts()

1    4592
0    1363
Name: none, dtype: int64

In [30]:
comment_issue_merge['mocking'].value_counts()

0    5775
1     180
Name: mocking, dtype: int64

In [31]:
comment_issue_merge['threat'].value_counts()

0    5932
1      23
Name: threat, dtype: int64

In [32]:
comment_issue_merge['vulgarity'].value_counts()

0    5884
1      71
Name: vulgarity, dtype: int64

In [33]:
comment_issue_merge['irony'].value_counts()

0    5891
1      64
Name: irony, dtype: int64

In [34]:

columns_to_drop = ['id_x', 'comment_body', 'id_y', 'consequences']

# Drop multiple columns
short_comment_issue_merge = comment_issue_merge.drop(columns=columns_to_drop)

In [35]:
short_comment_issue_merge

issue_id  comment_id                  comment_label    trigger  \
8       12894489    12894489                           None  Rejection   
9       12894489    19156555                           None  Rejection   
10      12894489   913841398                           None  Rejection   
11      12894489   913845504                          Irony  Rejection   
12      12894489   914030250                          Irony  Rejection   
...          ...         ...                            ...        ...   
5964  1958701035  1780979523  Identity attacks/Name-Calling  Rejection   
5965  1958701035  1958701035                           None  Rejection   
5966  1959812102  1777791158                           None       None   
5967  1959812102  1777796247                      Insulting       None   
5968  1959812102  1959812102                           None       None   

                    target  positive  negative  neutral consequence_category  \
8     Company/organization       0.0       1.0      0.0             negative   
9     Company/organization       0.0       1.0      0.0             negative   
10    Company/organization       0.0       1.0      0.0             negative   
11    Company/organization       0.0       1.0      0.0             negative   
12    Company/organization       0.0       1.0      0.0             negative   
...                    ...       ...       ...      ...                  ...   
5964  Company/organization       0.0       1.0      0.0             negative   
5965  Company/organization       0.0       1.0      0.0             negative   
5966                People       0.0       0.0      1.0              neutral   
5967                People       0.0       0.0      1.0              neutral   
5968                People       0.0       0.0      1.0              neutral   

      entitlement  identity  impatience  insulting  none  irony  mocking  \
8               0         0           0          0     1      0        0   
9               0         0           0          0     1      0        0   
10              0         0           0          0     1      0        0   
11              0         0           0          0     0      1        0   
12              0         0           0          0     0      1        0   
...           ...       ...         ...        ...   ...    ...      ...   
5964            0         1           0          0     0      0        0   
5965            0         0           0          0     1      0        0   
5966            0         0           0          0     1      0        0   
5967            0         0           0          1     0      0        0   
5968            0         0           0          0     1      0        0   

      threat  vulgarity  
8          0          0  
9          0          0  
10         0          0  
11         0          0  
12         0          0  
...      ...        ...  
5964       0          0  
5965       0          0  
5966       0          0  
5967       0          0  
5968       0          0  

[5955 rows x 18 columns]

In [36]:
# Entitlement
perc_entitlement = short_comment_issue_merge.groupby('issue_id')['entitlement'].mean() * 100
short_comment_issue_merge = short_comment_issue_merge.merge(perc_entitlement, on='issue_id', how='left')
short_comment_issue_merge.rename(columns={'entitlement_y': 'perc_entitlement'}, inplace=True)

# Identity
perc_identity = short_comment_issue_merge.groupby('issue_id')['identity'].mean() * 100
short_comment_issue_merge = short_comment_issue_merge.merge(perc_identity, on='issue_id', how='left')
short_comment_issue_merge.rename(columns={'identity_y': 'perc_identity'}, inplace=True)

# Impatience
perc_impatience = short_comment_issue_merge.groupby('issue_id')['impatience'].mean() * 100
short_comment_issue_merge = short_comment_issue_merge.merge(perc_impatience, on='issue_id', how='left')
short_comment_issue_merge.rename(columns={'impatience_y': 'perc_impatience'}, inplace=True)

# Insulting
perc_insulting = short_comment_issue_merge.groupby('issue_id')['insulting'].mean() * 100
short_comment_issue_merge = short_comment_issue_merge.merge(perc_insulting, on='issue_id', how='left')
short_comment_issue_merge.rename(columns={'insulting_y': 'perc_insulting'}, inplace=True)

# None
perc_none = short_comment_issue_merge.groupby('issue_id')['none'].mean() * 100
short_comment_issue_merge = short_comment_issue_merge.merge(perc_none, on='issue_id', how='left')
short_comment_issue_merge.rename(columns={'none_y': 'perc_none'}, inplace=True)

# Irony
perc_irony = short_comment_issue_merge.groupby('issue_id')['irony'].mean() * 100
short_comment_issue_merge = short_comment_issue_merge.merge(perc_irony, on='issue_id', how='left')
short_comment_issue_merge.rename(columns={'irony_y': 'perc_irony'}, inplace=True)

# Mocking
perc_mocking = short_comment_issue_merge.groupby('issue_id')['mocking'].mean() * 100
short_comment_issue_merge = short_comment_issue_merge.merge(perc_mocking, on='issue_id', how='left')
short_comment_issue_merge.rename(columns={'mocking_y': 'perc_mocking'}, inplace=True)

# Threat
perc_threat = short_comment_issue_merge.groupby('issue_id')['threat'].mean() * 100
short_comment_issue_merge = short_comment_issue_merge.merge(perc_threat, on='issue_id', how='left')
short_comment_issue_merge.rename(columns={'threat_y': 'perc_threat'}, inplace=True)

# Vulgarity
perc_vulgarity = short_comment_issue_merge.groupby('issue_id')['vulgarity'].mean() * 100
short_comment_issue_merge = short_comment_issue_merge.merge(perc_vulgarity, on='issue_id', how='left')
short_comment_issue_merge.rename(columns={'vulgarity_y': 'perc_vulgarity'}, inplace=True)


In [37]:
short_comment_issue_merge.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5955 entries, 0 to 5954
Data columns (total 27 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   issue_id              5955 non-null   int64  
 1   comment_id            5955 non-null   int64  
 2   comment_label         5955 non-null   object 
 3   trigger               5955 non-null   object 
 4   target                5955 non-null   object 
 5   positive              5955 non-null   float64
 6   negative              5955 non-null   float64
 7   neutral               5955 non-null   float64
 8   consequence_category  5955 non-null   object 
 9   entitlement_x         5955 non-null   int64  
 10  identity_x            5955 non-null   int64  
 11  impatience_x          5955 non-null   int64  
 12  insulting_x           5955 non-null   int64  
 13  none_x                5955 non-null   int64  
 14  irony_x               5955 non-null   int64  
 15  mocking_x            

In [38]:
# List of columns to keep
columns_to_keep = [
    'issue_id', 
    'trigger', 
    'target', 
    'consequence_category',
    'perc_entitlement',
    'perc_identity',
    'perc_impatience',
    'perc_insulting',
    'perc_none',
    'perc_irony',
    'perc_mocking',
    'perc_threat',
    'perc_vulgarity'
]

# Create the new DataFrame
condensed_issue_lvl = short_comment_issue_merge[columns_to_keep]

condensed_issue_lvl


issue_id    trigger                target consequence_category  \
0       12894489  Rejection  Company/organization             negative   
1       12894489  Rejection  Company/organization             negative   
2       12894489  Rejection  Company/organization             negative   
3       12894489  Rejection  Company/organization             negative   
4       12894489  Rejection  Company/organization             negative   
...          ...        ...                   ...                  ...   
5950  1958701035  Rejection  Company/organization             negative   
5951  1958701035  Rejection  Company/organization             negative   
5952  1959812102       None                People              neutral   
5953  1959812102       None                People              neutral   
5954  1959812102       None                People              neutral   

      perc_entitlement  perc_identity  perc_impatience  perc_insulting  \
0                  0.0            0.0              0.0        0.000000   
1                  0.0            0.0              0.0        0.000000   
2                  0.0            0.0              0.0        0.000000   
3                  0.0            0.0              0.0        0.000000   
4                  0.0            0.0              0.0        0.000000   
...                ...            ...              ...             ...   
5950               0.0           20.0              0.0        0.000000   
5951               0.0           20.0              0.0        0.000000   
5952               0.0            0.0              0.0       33.333333   
5953               0.0            0.0              0.0       33.333333   
5954               0.0            0.0              0.0       33.333333   

      perc_none  perc_irony  perc_mocking  perc_threat  perc_vulgarity  
0     50.000000        50.0           0.0          0.0             0.0  
1     50.000000        50.0           0.0          0.0             0.0  
2     50.000000        50.0           0.0          0.0             0.0  
3     50.000000        50.0           0.0          0.0             0.0  
4     50.000000        50.0           0.0          0.0             0.0  
...         ...         ...           ...          ...             ...  
5950  60.000000         0.0           0.0          0.0            20.0  
5951  60.000000         0.0           0.0          0.0            20.0  
5952  66.666667         0.0           0.0          0.0             0.0  
5953  66.666667         0.0           0.0          0.0             0.0  
5954  66.666667         0.0           0.0          0.0             0.0  

[5955 rows x 13 columns]

In [39]:
condensed_issue_lvl = condensed_issue_lvl.drop_duplicates()
condensed_issue_lvl

issue_id                                    trigger  \
0       12894489                                  Rejection   
6       13258430                                 Unprovoked   
33      22814728                     Technical disagreement   
37      24996968                                 Unprovoked   
50      27120442                     Technical disagreement   
...          ...                                        ...   
5936  1897778633                                       None   
5938  1906791437  Failed use of tool/code or error messages   
5940  1943190932                     Technical disagreement   
5947  1958701035                                  Rejection   
5952  1959812102                                       None   

                    target consequence_category  perc_entitlement  \
0     Company/organization             negative               0.0   
6                Code/tool              neutral               0.0   
33                  People              neutral               0.0   
37                  People              neutral               0.0   
50               Code/tool             positive               0.0   
...                    ...                  ...               ...   
5936                People              neutral               0.0   
5938             Code/tool              neutral               0.0   
5940                People              neutral               0.0   
5947  Company/organization             negative               0.0   
5952                People              neutral               0.0   

      perc_identity  perc_impatience  perc_insulting   perc_none  perc_irony  \
0              0.00         0.000000        0.000000   50.000000   50.000000   
6              0.00         7.407407        0.000000   77.777778    3.703704   
33             0.00        25.000000        0.000000   50.000000    0.000000   
37             0.00         0.000000        0.000000  100.000000    0.000000   
50             6.25         6.250000        0.000000   87.500000    0.000000   
...             ...              ...             ...         ...         ...   
5936           0.00         0.000000        0.000000   50.000000    0.000000   
5938           0.00         0.000000        0.000000   50.000000    0.000000   
5940           0.00        42.857143        0.000000   28.571429    0.000000   
5947          20.00         0.000000        0.000000   60.000000    0.000000   
5952           0.00         0.000000       33.333333   66.666667    0.000000   

      perc_mocking  perc_threat  perc_vulgarity  
0         0.000000          0.0        0.000000  
6         0.000000          0.0        3.703704  
33       25.000000          0.0        0.000000  
37        0.000000          0.0        0.000000  
50        0.000000          0.0        0.000000  
...            ...          ...             ...  
5936      0.000000          0.0        0.000000  
5938      0.000000          0.0        0.000000  
5940     28.571429          0.0        0.000000  
5947      0.000000          0.0       20.000000  
5952      0.000000          0.0        0.000000  

[403 rows x 13 columns]

In [40]:
condensed_issue_lvl.to_stata('condensed_issue_lvl.dta', write_index=False)

In [41]:
short_comment_issue_merge['consequence_category'].value_counts()

neutral     2768
negative    1542
positive    1202
balanced     443
Name: consequence_category, dtype: int64

In [42]:
condensed_issue_lvl['consequence_category'].value_counts()

neutral     244
negative     88
positive     55
balanced     16
Name: consequence_category, dtype: int64

In [43]:
condensed_issue_lvl['target'].value_counts()

People                  167
None                    113
Code/tool                77
Company/organization     32
Undirected               14
Name: target, dtype: int64

In [44]:
condensed_issue_lvl['trigger'].value_counts()

None                                         124
Failed use of tool/code or error messages     72
Technical disagreement                        61
Communication breakdown                       52
Unprovoked                                    30
Politics/ideology                             24
Rejection                                     15
Past interactions                             13
Violation of community conventions            12
Name: trigger, dtype: int64